network-mux/src/Network/Mux/Types.hs

```diff
@@ -66,6 +70,23 @@ newtype RemoteClockModel
 remoteClockPrecision :: DiffTime
 remoteClockPrecision = 1e-6
 
+-- May want to make this a configuration variable
+
+-- NOTE this interval must be less than the wrap around time of the
+-- `RemoteClockModel`. The remote clock model has a precision of
+-- `remoteClockPrecision`.
+sampleIntervalPicoseconds :: Q Exp
+sampleIntervalPicoseconds = check 10
+  where
+    check n
+     | n > 0 && n < wrapInterval
+       = let picos = diffTimeToPicoseconds n
+         in  [|picos|]
+     | otherwise
+       = fail $ "sampleIntervalPicoseconds: infeasible sampleInterval: "<>show n
+    wrapInterval
+      = remoteClockPrecision * (fromIntegral $ unRemoteClockModel maxBound)
```

Comment:
I've used Template Haskell to lift computation (or failure to compute) of this constant to compile time. Is this too much, do you think?